File: widgetssdk/src/main/java/com/glia/widgets/chat/model/history/ResponseCardItem.java
package com.glia.widgets.chat.model.history;

import androidx.annotation.NonNull;

import com.glia.androidsdk.chat.SingleChoiceOption;

import java.util.Collections;
import java.util.List;
import java.util.Objects;

public class ResponseCardItem extends OperatorMessageItem {
    @NonNull
    public final List<SingleChoiceOption> singleChoiceOptions;
    public final String choiceCardImageUrl;

    public ResponseCardItem(
            @NonNull String id,
            String operatorName,
            String operatorProfileImgUrl,
            boolean showChatHead,
            String content,
            String operatorId,
            long timestamp,
            //Must not be empty, otherwise it is OperatorMessageItem
            @NonNull List<SingleChoiceOption> singleChoiceOptions,
            String choiceCardImageUrl
    ) {
        super(id, operatorName, operatorProfileImgUrl, showChatHead, content, operatorId, timestamp);

        assert !singleChoiceOptions.isEmpty();
        this.singleChoiceOptions = Collections.unmodifiableList(singleChoiceOptions);
        this.choiceCardImageUrl = choiceCardImageUrl;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof ResponseCardItem)) return false;
        if (!super.equals(o)) return false;
        ResponseCardItem that = (ResponseCardItem) o;
        return singleChoiceOptions.equals(that.singleChoiceOptions) && Objects.equals(choiceCardImageUrl, that.choiceCardImageUrl);
    }

    @Override
    public int hashCode() {
        return Objects.hash(super.hashCode(), singleChoiceOptions, choiceCardImageUrl);
    }

    @NonNull
    @Override
    public String toString() {
        return "ResponseCardItem{" +
                "singleChoiceOptions=" + singleChoiceOptions +
                ", choiceCardImageUrl='" + choiceCardImageUrl + '\'' +
                "} " + super.toString();
    }
}

File: widgetssdk/src/main/java/com/glia/widgets/chat/model/history/ResponseCardItem.kt
package com.glia.widgets.chat.model.history

import com.glia.androidsdk.chat.SingleChoiceOption

data class ResponseCardItem(
    private val id: String,
    private val operatorName: String?,
    private val operatorProfileImgUrl: String?,
    private val showChatHead: Boolean,
    private val content: String?,
    private val operatorId: String?,
    private val timestamp: Long,
    val singleChoiceOptions: List<SingleChoiceOption>,
    val choiceCardImageUrl: String?
) : OperatorMessageItem(id, operatorName, operatorProfileImgUrl, showChatHead, content, operatorId, timestamp) {

    init {
        require(singleChoiceOptions.isNotEmpty()) { "Response card should have at least one `SingleChoiceOption`" }
    }
}
